### PR TITLE
fix: 🐛 Phone Mask - allows 1's in fourth position

### PR DESCRIPTION
**Pull request by hvilander**
The phone mask in a masked text field was too aggressive, in that it
would delete any 1's that ended up in the fourth position. This change
fixes that and adds some documentation.

Loom: https://www.loom.com/share/840da841411545be9d6a928cf38e81cd

✅ Closes: #392

### CODE REVIEW

**Comment by 20BBrown14** [comment]
Hm. I guess I'm a little wary about where the line is for masks that "ensure validity of the data" and "ensure proper format of the data"? If a phone number isn't valid with a `1` in the 4th position then it should never come about that it's an "issue". If that makes sense.

Was there something that sparked this change?

**Comment by 20BBrown14** [comment]
> Hm. I guess I'm a little wary about where the line is for masks that "ensure validity of the data" and "ensure proper format of the data"? If a phone number isn't valid with a `1` in the 4th position then it should never come about that it's an "issue". If that makes sense.
> 
> Was there something that sparked this change?

Aight, I got called out. I didn't watch the loom before commenting this but then loom was great and answered my question. I can see why. Our library isn't created to make sure data in the forms is valid for whatever use case consumers might need it for. The library is just here to facilitate displaying and collecting that data in an easy to understand and use way.

File: SQFormDocs/docs/masked_text_field.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+## Masks
+
+For convenience several common masks are provided. `import MASKS from '
+
+- phone: US Phone numbers \*\*Will allow invalid phone numbers see note below
+- zip: US address zip codes
+- currency: a number only mask that allows decimal values.
+- percent: a number mask that adds a `%` as a suffix
+- email: provided from the [text-mask module](https://github.com/text-mask/text-mask/tree/master/addons/#emailmask)
+- date: a mask to allow the date form of 12/31/2020.
```
Comment (20BBrown14) [inline SQFormDocs/docs/masked_text_field.md:39]: Would it be better to use `MM/DD/YYYY` in this doc?

**Review by 20BBrown14 — APPROVED** [review]
Approved pending small changes

File: SQFormDocs/docs/masked_text_field.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+| size | 'auto' or a number 1 - 12 | no | Size of the input given full-width is 12. |
+| onBlur | function | no | Custom onBlur event callback |
+| onChange | function | no | Custom onChange event callback |
+| startAdornment | node | no | Adornment that appears at the start of the input |
+| endAdornment | node | no | Adornment that appears at the end of the input |
+| type | string | no | Defines the input type for the text field. Must be a valid HTML5 input type |
+| InputProps | object | no | Props applied to the Input element |
+| inputProps | object | no | Attributes applied to the `input` element |
+| muiFieldProps | object | no | Any valid prop for MUI text input child component https://material-ui.com/api/text-field/#props |
+| strpNonNumeric | boolean | no | The submitted value from the input will have all non-numeric characters removed |
```
[the remark this answers is not in the document]
Comment (hvilander) [reply SQFormDocs/docs/masked_text_field.md:28]: good catch!

File: SQFormDocs/docs/masked_text_field.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+## Masks
+
+For convenience several common masks are provided. `import MASKS from '
```
[the remark this answers is not in the document]
Comment (hvilander) [reply SQFormDocs/docs/masked_text_field.md:32]: This was meant to be an example I apparently lost along the way. I will add it in as a separate section.

File: SQFormDocs/docs/masked_text_field.md (hunk 1 above)
Comment (hvilander) [reply SQFormDocs/docs/masked_text_field.md:39]: It absolutely would.

**Pull request merged**